release_dashboard/lib/state/status_state.dart

```diff
@@ -30,6 +30,26 @@ class StatusState extends ChangeNotifier {
 
     notifyListeners();
   }
+
+  /// Updates the release status with the latest values saved in the state file.
+  ///
+  /// Use the code below to call the method:
+  ///
+  /// {@tool snippet}
+  ///
+  /// [context] is the [BuildContext] of the widget which is calling this method.
+  /// [read] is from 'package:provider/provider.dart'.
+  ///
+  /// An example on how to call this method:
+  ///
+  /// ```dart
+  /// context.read<StatusState>().syncStatusWithState();
+  /// ```
+  /// {@end-tool}
+  void syncStatusWithState() {
+    releaseStatus = stateToMap(conductor.state);
+    notifyListeners();
```

Comment:
There should be wrapper calls in the state of accessing the direct conductor instance from state. This class should have functions like

`createRelease` and `cleanContext` as those will trigger `notifyListeners` when ready. As it is, we're forced to always call `syncStatusWithState` which requires the widgets to be depending on the business logic of conductor, over complicating the UI.

Reply:
Could you provide some examples, please?

Reply:
```dart
// status_state.dart

Future<void> cleanContext() async {
  await _runConductorAction(this._conductor.cleanContext);
}

Future<void> _runConductorAction(DynamicFuture conductorFuture) {
  try {
    await _conductorFuture();
  } catch (ConductorException exception) {
    _error = exception;
  } catch (exception) {
    print(exception);
    exit(1);
  } finally {
    notifyListeners();
  }
}
```

Reply:
The conductor core offers a method `updateState` for `StartContext` and `NextContext` that could be overriden. This method is called whenever the core changes the state file in disk, and that's when the release state in the release dashboard needs to be synced. The sync for `StartContext` for example happens here [here](https://github.com/Yugue/cocoon/blob/d91e2518a90426408551acc1648bb179a12591ae/release_dashboard/lib/services/release_dashboard_start_context.dart#L53).

 `syncStatusWithState` needs only to be called in the override `updateState` of `StartContext` and `NextContext`.
 With the above code, do I still need to call `updateState` in the override?

Reply:
The services shouldn't be responsible for updating the UI. That's what the state file is for

Service <=> State <=> UI

What you've mentioned is Service <=> UI which is indicating we're passing around the conductor service which shouldn't be happening anymore.

Reply:
The state files are only calling `syncStatusWithState` which is a function of `state` file. Apart from that, the service files do not have any interaction with the UI. Isn't the relation `Service <=> State <=> UI` respected?

Reply:
I added a todo [here](https://github.com/Yugue/cocoon/blob/f0e2b167044ee45d26ed5bbd533d7e40f863fccd/release_dashboard/lib/state/status_state.dart#L54) for the extra layer of abstraction between service and state, and the todo links to this conversation for context. Since it requires a significant amount of code refactoring in the `NextContext` and `StartContext` services, it will be handled in a future PR.